messaging/src/main/java/org/axonframework/messaging/annotation/HandlerAttributes.java

```diff
@@ -0,0 +1,144 @@
+package org.axonframework.messaging.annotation;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Container for message handler attributes. Can store several {@link Map}s of {@link String} to {@link Object} under
+ * their own key, resembling a message handling method which serves several handling functions.
+ *
+ * @author Steven van Beelen
+ * @since 4.5
+ */
+public class HandlerAttributes {
+
+    private final Map<String, Map<String, Object>> attributes;
+
+    /**
+     * Constructs an empty handler attributes object.
+     */
+    public HandlerAttributes() {
+        this(new HashMap<>());
+    }
+
+    /**
+     * Construct a handler attributes object based on the given {@code handlerAttributes} {@link Map}.
+     *
+     * @param attributes a {@link Map} from handler type to attributes {@code Map} to base a {@link HandlerAttributes}
+     *                   on
+     */
+    public HandlerAttributes(Map<String, Map<String, Object>> attributes) {
```

Comment:
Should this just be key->value mapping, instead of key->key->value?

Reply:
My initial idea was "no", but let's adjust the API more thoroughly by no longer providing a means to retrieve all attributes for a given handler type. With that in place, I'd say "yes" to this.